app/models/invenio_rdm_record_converter.rb

```diff
@@ -227,10 +227,10 @@ def build_creator_contributor_json(creator)
           end
       }
     # User within DigitalHub
-    elsif dh_user = User.find_by(formal_name: creator)
+    elsif dh_user = User.find_by(formal_name: creator) || dh_user = User.find_by(display_name: creator.split(", ").reverse.join(" ").strip)
```

Comment:
Kind of a busy line. All this does is look a user up by reversing the formal name and remove the comma

Reply:
Kind of a busy *method* :wink:  . Free text is gnarly. We can corral it though. From the outside it's hard to tell if it does what it's supposed to do because cases are a bit hidden and there is a lot of repeated tweaking.

Here are a couple of suggestions:

- We can remove `Hash.new.tap do` pattern and just use `{}` directly. `{}` is already used at the top-level of these hashes and since no operations are executed, tapping has little value (readability is already accounted for by the hash indent)
- Split the if/elsif statements to isolate the first case. It's just a cache and doesn't deal with creating a new person_or_org. Smthg like 
```
json = @@person_or_org_data[creator]
return json if json

# Create json since not found
if organization?(creator)  
```
- in places where split may lead to nils, do the string conversion right away, so that repeated `&.` and `to_s` are not necessary. (e.g., 
```ruby
family_name, given_name = creator.split(',', 2)
given_name = given_name.to_s.lstrip
family_name = family_name.to_s.lstrip

if !(given_name =~ /\d/)
  given_name = given_name.gsub(',', "")
end
# ...
```
- add comments that show the cases being tackled (e.g. `# This accounts for user with date pattern e.g. "Mason, Michael L., 1895-1963"` or something higher-level)
- replace the busy line by a function (there is probably a way to make those 2 DB calls into one with an Active(Record/Fedora) `OR` expression )
- Don't see how the following comment "remove any commas that are not next to a "."" matches with the code. Not applicable anymore ?
- Replace the repeated `json = @@person_or_org_data[creator] =` by `json =` and a single `@@person_or_org_data[creator] = json` at the end .